### PR TITLE
Post about verifying bundles from various services with cosign

### PR DESCRIPTION
**Pull request by steiza**
#### Summary

There's some really cool features in the cosign v2.4.0 release that allows you to verify attestations in the bundle format used by npm provenance, GitHub Artifact Attestations, and Homebrew provenance. Let's show people how it works!

#### Release Note

NONE

#### Documentation

N/A

### CODE REVIEW

File: content/cosign-verify-bundles.md
Comment (haydentherapper) [inline content/cosign-verify-bundles.md:33]: Can we call out `--new-bundle-format` explicitly as the addition that makes it possible to support this new bundle format?

File: content/cosign-verify-bundles.md
Comment (haydentherapper) [inline content/cosign-verify-bundles.md:35]: Can we note that the blob's digest is also checked to be one of the subjects of the attestation? (`--check-claims` is true by default)

File: content/cosign-verify-bundles.md
Comment (haydentherapper) [inline content/cosign-verify-bundles.md:74]: Do you also need `--insecure-ignore-tlog`, since there is no Rekor proof? You can also use `--private-infrastructure`, these two flags are equivalent.

File: content/cosign-verify-bundles.md
Comment (steiza) [reply content/cosign-verify-bundles.md:24]: Yes! But I'm going to link it above where we first mention bundles.

File: content/cosign-verify-bundles.md
Comment (steiza) [reply content/cosign-verify-bundles.md:35]: Technically this is using the sigstore-go verification path, so the claims will always be checked. But I did add a paragraph describing (at a high level!) what's being verified.

File: content/cosign-verify-bundles.md
Comment (steiza) [reply content/cosign-verify-bundles.md:76]: We don't in this case since we are providing `--use-signed-timestamps`. If we did *not* have `--use-signed-timestamps`, we *would* need `--ignoreTlog`.

This was an implementation choice on how to map cosign flags to sigstore-go verifier config: https://github.com/sigstore/cosign/blob/b23586d6390d6a48ba4789848fe6ad89710afb7f/cmd/cosign/cli/verify/verify_bundle.go#L141-L147

I'll be the first to admit that this is a bit messy, and am open to feedback here.

Comment (haydentherapper) [reply content/cosign-verify-bundles.md:76]: Whoops sorry, forgot this was a different path!

For https://github.com/sigstore/cosign/blob/b23586d6390d6a48ba4789848fe6ad89710afb7f/cmd/cosign/cli/verify/verify_bundle.go#L133-L135, I think we should remove `len(trustedroot.RekorLogs())` and require that you set a flag to skip the log check, otherwise a malicious can just remove the log entries from the bundle to skip over the check.

Comment (steiza) [reply content/cosign-verify-bundles.md:76]: I think this is okay - note that we're checking to see if there's Rekor entries in the trusted root, *not* the bundle.

For better or worse, we put a lot of trust in the supplied trusted root today (which is why we need to work towards having cosign retrieve it via TUF in more cases!)

Comment (haydentherapper) [reply content/cosign-verify-bundles.md:76]: Oh you’re totally right, ignore this! Thanks

**Review by haydentherapper — DISMISSED** [review]

**Review by trevrosen — DISMISSED** [review]
A couple stylistic things - overall I like the direction and I'll let folks like @haydentherapper and @bobcallaway do more technical review.

:shipit:

**Review by bobcallaway — DISMISSED** [review]

**Comment by tracymiranda** [comment]
Great post. 
One suggestion to add a sentence early on that emphasizes that the intent is for all clients to standardize on the bundle format, as per roadmap: https://github.com/sigstore/community/blob/main/ROADMAP.md#client-sdks
Thanks for putting this together!

**Review by haydentherapper — APPROVED** [review]

**Review by SantiagoTorres — APPROVED** [review]
LGTM. I wonder if we want to mention that, in practice, you may not need to embed the bundle policy in cosign itself, and that higher-level tools can figure this out for you.
Those command lines are a bit long :sweat_smile:

File: content/cosign-verify-bundles.md
Comment (bobcallaway) [inline content/cosign-verify-bundles.md:4]: you might want to update this before publishing

Comment (steiza) [reply content/cosign-verify-bundles.md:4]: I feel you, but that will again dismiss all the reviews. Is this something a maintainer can do before landing? Or should I update it and hope we get timely re-reviews?

Comment (SantiagoTorres) [reply content/cosign-verify-bundles.md:4]: I think you're fine. I don't see any outstanding comments, anyway (mine is a suggestion :))

Comment (bobcallaway) [reply content/cosign-verify-bundles.md:4]: if you have 3 TSC LGTMs, i'm happy to force merge for a trivial fix like setting a date.

**Review by trevrosen — APPROVED** [review]

**Pull request merged**
